views/components/colorPicker.jsx

```diff
@@ -0,0 +1,42 @@
+import { Paper, Typography } from "@material-ui/core"
+import React, { useCallback, useRef, useState } from "react"
+import { HexColorPicker } from "react-colorful"
+
+import useClickOutside from "../hooks/useClickOutside"
+
+const ColorPicker = ({ 
+  classes, 
+  color, 
+  onColorChange, 
+  idx
+}) => {
+  const [isColorPickerOpen, setColorPickerToggle] = useState(false)
+  const popover = useRef()
+  const close = useCallback(() => setColorPickerToggle(false), [])
+  useClickOutside(popover, close)
+
+  return (
+    <div className={classes.swatchContainer}>
+      {!isColorPickerOpen &&(
+        <>    
+          <Typography variant='caption' className={classes.colorLabel}>
+            Color
+          </Typography>
+          <Paper
+            className={classes.swatch}
+            style={{ backgroundColor: color }}
+            onClick={() => setColorPickerToggle(true)}
+          />
+        </>
+
+      )}
+      {isColorPickerOpen && (
+        <div className={classes.popover} ref={popover}>
+          <HexColorPicker color={color} onChange={(newColor) => onColorChange({target: { name: 'color', value: newColor }}, idx)} />
+        </div>
+      )}
```

Comment:
nitpick: Adding some sort of placeholder when the color picker is open would be a nice-to-have. Currently, the little color swatch square disappears when the color picker is opened. I wouldn't call this a blocker, but it might be something worth noting for future work.

Reply:
https://drive.google.com/file/d/1Li1xAaoMnSkfcwJjVkqLYH8vPhPlpUIE/view?usp=sharing

Reply:
Thanks for making that change. Looks great!